FIAT/check_format_variant.py

```diff
@@ -0,0 +1,32 @@
+import re
+import warnings
+
+
+def check_format_variant(variant, degree, element):
+    if variant is None:
+        variant = "point"
+        warnings.simplefilter('always', DeprecationWarning)
+        warnings.warn('Variant of ' + element + ' element will change from point evaluation to integral evaluation.'
+                      ' You should project into variant="integral"', DeprecationWarning)
+        # Replace by the following in a month time
+        # variant = "integral"
+
+    if not (variant == "point" or "integral" in variant):
+        raise ValueError('Choose either variant="point" or variant="integral"'
+                         'or variant="integral(Quadrature degree)"')
+
+    if variant == "integral":
+        # quadrature is so high to ensure that the interpolant of curl/divergence-free functions is still curl/divergence-free
+        quad_deg = 5 * (degree + 1)
+        variant = "integral"
+    elif re.match(r'^integral\(\d+\)$', variant):
+        quad_deg = int(''.join(filter(str.isdigit, variant)))
```

Comment:
This is when I would like to be able to bind in tests. 

Prefer to just grab this match up front.

```python
match = re.match("^integral(?:\((\d+)\))?$", variant)
if match:
    quad_degree, = match.groups()
    quad_degree = int(quad_degree) if quad_degree is not None else 5*(degree + 1)
    if quad_deg < degree + 1:
        raise ValueError(...)
elif variant == "point":
    quad_deg = None
else:
    raise ValueError(message as above)
```

WDYT?